tools/testsys/src/crds.rs

```diff
@@ -159,13 +215,110 @@ impl<'a> CrdInput<'a> {
+            "flavor".to_string() => some_or_null(&self.variant.variant_flavor().map(str::to_string)),
+            "version".to_string() => some_or_null(&self.variant.version().map(str::to_string)),
             "cluster-name".to_string() => cluster_name.to_string(),
             "instance-type".to_string() => some_or_null(&self.config.instance_type),
             "agent-role".to_string() => some_or_null(&self.config.agent_role),
             "conformance-image".to_string() => some_or_null(&self.config.conformance_image),
             "conformance-registry".to_string() => some_or_null(&self.config.conformance_registry),
         }
     }
+
+    /// Find the crd template file for the given test type
+    fn custom_crd_template_file_path(&self) -> Option<PathBuf> {
+        let test_type = &self.test_type.to_string();
+        // List all acceptable paths to the custom crd to allow users some freedom in the way
+        // `tests` is organized.
+        let acceptable_paths = vec![
+            // Check for <TEST-TYPE>.yaml in the top level directory
+            PathBuf::new().join(test_type).with_extension("yaml"),
+            // Check for <TESTSYS_FOLDER>/<TEST-TYPE>/<TEST-TYPE>.yaml
+            self.tests_directory
+                .join(test_type)
+                .join(test_type)
+                .with_extension("yaml"),
+            // Check for <TESTSYS_FOLDER>/<TEST-TYPE>/test.yaml
+            self.tests_directory.join(test_type).join("test.yaml"),
+            // Check for <TESTSYS_FOLDER>/<TEST-TYPE>/crd.yaml
+            self.tests_directory.join(test_type).join("crd.yaml"),
+            // Check for <TESTSYS_FOLDER>/<TEST-TYPE>/crds.yaml
+            self.tests_directory.join(test_type).join("crds.yaml"),
+            // Check for <TESTSYS_FOLDER>/shared/<TEST-TYPE>.yaml
+            self.tests_directory
+                .join("shared")
+                .join(test_type)
+                .with_extension("yaml"),
+            // Check for <TESTSYS_FOLDER>/shared/tests/<TEST-TYPE>.yaml
+            self.tests_directory
+                .join("shared")
+                .join("tests")
+                .join(test_type)
+                .with_extension("yaml"),
+        ];
+
+        // Find the first acceptable path that exists and return that.
+        acceptable_paths.into_iter().find(|path| path.exists())
+    }
+
+    /// Find the cluster config file for the given cluster name and test type.
+    fn cluster_config_file_path(&self, cluster_name: &str) -> Option<PathBuf> {
+        let test_type = &self.test_type.to_string();
+        // List all acceptable paths to the custom crd to allow users some freedom in the way
+        // `tests` is organized.
+        let acceptable_paths = vec![
+            // Check for <TESTSYS_FOLDER>/<TEST-TYPE>/<CLUSTER-NAME>.yaml
+            self.tests_directory
+                .join(test_type)
+                .join(cluster_name)
+                .with_extension("yaml"),
```

Comment:
There's some potential ambiguity here, e.g. if my cluster is named "quick" then we'd resolve the same file for both the cluster config and the CRD template.

What do you think about enforcing a convention that encodes the file type in the name, like `<CLUSTER-NAME>.cluster.yaml` vs `<CLUSTER-NAME>.test.yaml`?

I'm a bit sad that there's no orthogonal case to `<TESTSYS_FOLDER>/<TEST-TYPE>/crd.yaml` here; could we also support `<TESTSYS_FOLDER>/<TEST-TYPE>/cluster.yaml`, or does that not make sense for some reason?

Reply:
`<TESTSYS_FOLDER>/<TEST-TYPE>/cluster.yaml` doesn't work because we support run a test on multiple clusters.

Reply:
The name of the cluster config does not have the be the name of the cluster. If you wanted a cluster named quick, you could name the cluster config `quick-cluster.yaml` to avoid the collision.